app/models/user_data.rb

```diff
@@ -0,0 +1,5 @@
+class UserData < ApplicationRecord
```

Comment:
+1 to what Ryan said.

Looking at the [entity relationship diagram](https://user-images.githubusercontent.com/94846816/262934836-076c4e4d-7f9d-439b-9fb3-fedadfbf6295.jpg) that you shared, which I realise covers more scope than just this PR, it did make me wonder:

What is the distinction between a `User` and a `UserData`? For example, why would the user's `staff_number` belong to one object, but their `years_of_service` belong to the other? The two seem very tightly coupled, which sort of indicates that they might not warrant being separate entities – it feels like they represent two halves of the same 'thing'.

It might be worth considering a couple of other options:

1. **Store all these fields on the `User` model.** So a user would have `annual_leave_remaining` directly on the `User` model.
2. **Define a clearer distinction between the two models.** For example, you might decide that the purpose of the `User` object is solely to facilitate users logging in to the app (e.g. it holds their login credentials, password reset tokens, and so on), and then have a `StaffMember` or `Employee` model to hold details specific to that user's employment (e.g. their staff number, years of service, line manager, remaining annual leave, etc.)

Option 2 might better support a future use-case where Staff Members could have multiple User accounts, or even to support different types of Users who can login but are not Staff Members. However if you [don't think you'll need that](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it), then Option 1 could be the simpler approach.

Happy to talk it through further if you want to! 🙂

Reply:
After extensive talks with Ollie, I've gone down the route of just having a `User` table and model and having everything contained in there (see new class diagram below). It is the assumption for the time being that all users will be employees, if this changes down the line (e.g. some users are not employees who will use the tool) then the annual leave data can be extracted into another class and migrated to another table, so this is a fairly low-risk assumption.

![class diagram (3)](https://github.com/jyoung-gds/govuk-holiday-logger/assets/94846816/9b4a7bb3-d4fd-4039-85dd-f25c86776cb2)